### ISSUE TITLE
Python3 (probably Python also): Disappearing child nodes

### ISSUE THREAD

**Issue by ronald-d-rogers** [antlr/antlr3#105]
Note I do not know if this is an intended change or a bug.
Also not very familiar w/ github and my first issue report so forgive me if I do anything retarded.

Using latest snapshot ([16284da443](https://github.com/antlr/antlr3/commit/16284da4436bcb57f339db01e81c4bea94663fff))

Child nodes are dropped in calls to method `nextNode()` in class `RewriteRuleSubtreeStream` after commit [3bfbd9c9cd](https://github.com/antlr/antlr3/blob/511919a34a3f2475efd067cfa2c30ca204e11987/runtime/Python/antlr3/tree.py#L2767) when a child tree is added to a parent tree that already has children.

Grammar:

```
grammar ClassSchemaPath;

options {
    output=AST;
    ASTLabelType=CommonTree;
    language = Python3;
}

tokens {
    ROOT;
    ELEM;
    ATTR;
    MULT;
    VAL;
}

/*------------------------------------------------------------------
 * PARSER RULES
 *------------------------------------------------------------------*/

parse: sibs EOF -> ^(ROOT sibs) ;

sibs: kids (SIB_SEP! kids)* ;

kids: (a=atom -> $a) (CH_SEP (k=kids -> ^($a $k)))? ;

atom: elem
    | SIB_BEG! sibs SIB_END! ;

elem: NAME attribute* multiplier? -> ^(ELEM[$NAME] attribute* multiplier?) ;

attribute: ATTR_BEG NAME ATTR_EQ ATTR_VAL ATTR_END -> ^(ATTR[$NAME] VAL[$ATTR_VAL]) ;

multiplier: MULT_BEG MULT_VAL MULT_END -> ^(MULT['*'] VAL[$MULT_VAL]) ;

/*------------------------------------------------------------------
 * LEXER RULES
 *------------------------------------------------------------------*/

ATTR_EQ     : '=';
ATTR_VAL    : ( '"' (~'"')* '"'|'\'' (~'\'')* '\'');
MULT_VAL    : (DIGIT)+;
NAME        : (LETTER|'_') (NAMECHAR)*;
SIB_SEP     : ',';
CH_SEP      : '>';
SIB_BEG     : '(';
SIB_END     : ')';
ATTR_BEG    : '[';
ATTR_END    : ']';
MULT_BEG    : '{';
MULT_END    : '}';

WS      : (' '|'\t'|'\r'|'\n'|'\u000C')+ {self.skip()};

fragment
NAMECHAR    : LETTER | DIGIT | '-' | '_' ;

fragment
LETTER      : 'a'..'z'|'A'..'Z';

fragment
DIGIT       : '0'..'9';
```

For input:

```
child_0[name="value"]{3} > child_1 > (child_2_1, child_2_2)
```

Changes from output:

```
(ROOT (child_0 (name "value") (* 3) (child_1 child_2_1 child_2_2)))
```

To output:

```
(ROOT (child_0 (child_1 child_2_1 child_2_2)))
```

Where **child_0**'s children **(name "value") (\* 3)** are dropped.

Tracked it down to [this change](https://github.com/antlr/antlr3/commit/511919a34a3f2475efd067cfa2c30ca204e11987#L4R2794)

From:

``` python
        # test size above then fetch
        el = self._next()
        return el
```

To:

``` python
        # test size above then fetch
        el = self._next()
        # dup just the root (want node here)
        return self.adaptor.dupNode(el)
```

Basically when children are being added to a tree which already has children the existing children are dropped as with the case of adding tree **(child_1 child_2_1 child_2_2)** to parent **(child_0 (name "value") (\* 3))**.

Here the call to dupNode() causes a call to CommonTree( **(child_0 (name "value") (\* 3))** ), which returns only the parent's root node **child_0**, right before the generated parser appends to it the child tree **(child_1 child_2_1 child_2_2)**.

The issue disappears when the code is reverted and the parent tree is returned with children untouched.
